Spigot-Server-Patches/0299-Add-ability-to-not-load-default-worlds.patch

```diff
@@ -0,0 +1,615 @@
+From 102785899ed75f36cb2be4d0b763fb3f4a9082f4 Mon Sep 17 00:00:00 2001
+From: theminecoder <buisness@theminecoder.me>
+Date: Fri, 18 May 2018 17:31:47 +1000
+Subject: [PATCH] Add ability to not load default worlds
+
+This enables minigame-like servers that most of the time are loading pre-made worlds
+to not have the extra main world(s) loaded on top of their custom ones.
+
+When in this mode and no world are loaded by the time the server is loaded a new
+will print out to the console to make sure that people know what they are doing.
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+index b602bbf12..c8abecc15 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+@@ -236,7 +236,7 @@ public class PaperConfig {
+     private static void bungeeOnlineMode() {
+         bungeeOnlineMode = getBoolean("settings.bungee-online-mode", true);
+     }
+-    
++
+     public static int packetInSpamThreshold = 300;
+     private static void packetInSpamThreshold() {
+         if (version < 11) {
+@@ -285,10 +285,20 @@ public class PaperConfig {
+ 
+     public static boolean savePlayerData = true;
+     private static void savePlayerData() {
+-        savePlayerData = getBoolean("settings.save-player-data", savePlayerData);
+-        if(!savePlayerData) {
++        savePlayerData = loadDefaultWorlds && getBoolean("settings.save-player-data", savePlayerData);
```

Comment:
This doesn't feel intuitive to me that player data saving would be disabled for this.

I understand that data is saved in a worlds folder, but that is more an implementation detail to me.

I would rather see player save data location and enablement to be a separate patch / config.